code/modules/mob/living/carbon/human/human.dm

```diff
@@ -835,43 +835,75 @@
+		if(!src.is_busy && (src.zone_selected == BODY_ZONE_HEAD || src.zone_selected == BODY_ZONE_PRECISE_GROIN) && get_turf(src) == get_turf(T) && !(T.mobility_flags & MOBILITY_STAND) && src.a_intent != INTENT_HELP) //all the stars align, time to curbstomp
 			src.is_busy = TRUE
 
-			if (!do_mob(src,T,25) || src.zone_selected != BODY_ZONE_HEAD || get_turf(src) != get_turf(T) || (T.mobility_flags & MOBILITY_STAND) || src.a_intent == INTENT_HELP) //wait 30ds and make sure the stars still align
+			if (!do_mob(src,T,25) || get_turf(src) != get_turf(T) || (T.mobility_flags & MOBILITY_STAND) || src.a_intent == INTENT_HELP) //wait 30ds and make sure the stars still align (Body zone check removed after PR #958)
 				src.is_busy = FALSE
 				return
 
 			T.Stun(6)
 
-			var/increment = (T.lying/90)-2
-			setDir(increment > 0 ? WEST : EAST)
-			for(var/i in 1 to 5)
-				src.pixel_y += 8-i
-				src.pixel_x -= increment
-				sleep(0.2)
-			for(var/i in 1 to 5)
-				src.pixel_y -= 8-i
-				src.pixel_x -= increment
-				sleep(0.2)
-
-			playsound(src, 'sound/effects/hit_kick.ogg', 80, 1, -1)
-			playsound(src, 'sound/weapons/punch2.ogg', 80, 1, -1)
-
-			var/obj/item/bodypart/BP = T.get_bodypart(BODY_ZONE_HEAD)
-			if(BP)
-				BP.receive_damage(36) //so 3 toolbox hits
-
-			T.visible_message("<span class='warning'>[src] curbstomps [T]!</span>", "<span class='warning'>[src] curbstomps you!</span>")
+			if(src.zone_selected == BODY_ZONE_HEAD) //curbstomp specific code
+
+				var/increment = (T.lying/90)-2
+				setDir(increment > 0 ? WEST : EAST)
+				for(var/i in 1 to 5)
+					src.pixel_y += 8-i
+					src.pixel_x -= increment
+					sleep(0.2)
+				for(var/i in 1 to 5)
+					src.pixel_y -= 8-i
+					src.pixel_x -= increment
+					sleep(0.2)
+
+				playsound(src, 'sound/effects/hit_kick.ogg', 80, 1, -1)
+				playsound(src, 'sound/weapons/punch2.ogg', 80, 1, -1)
+
+				var/obj/item/bodypart/BP = T.get_bodypart(BODY_ZONE_HEAD)
+				if(BP)
+					BP.receive_damage(36) //so 3 toolbox hits
+				
+				T.visible_message("<span class='warning'>[src] curbstomps [T]!</span>", "<span class='warning'>[src] curbstomps you!</span>")
+
+				log_combat(src, T, "curbstomped")
+
+			else if(src.zone_selected == BODY_ZONE_PRECISE_GROIN) //groinkick specific code
+
+				var/increment = (T.lying/90)-2
+				setDir(increment > 0 ? WEST : EAST)
+				for(var/i in 1 to 5)
+					src.pixel_y += 2-i
+					src.pixel_x -= increment
+					sleep(0.2)
+				for(var/i in 1 to 5)
+					src.pixel_y -= 2-i
+					src.pixel_x -= increment
+					sleep(0.2)
+
+				playsound(src, 'sound/effects/hit_kick.ogg', 80, 1, -1)
+				playsound(src, 'sound/effects/hit_punch.ogg', 80, 1, -1)
+
+				var/obj/item/bodypart/BP = T.get_bodypart(BODY_ZONE_CHEST)
+				if(T.gender == MALE)
+					if(BP)
+						BP.receive_damage(25)
+				else
+					if(BP)
+						BP.receive_damage(15)

```

Comment:
Like I already said, gender check should go inside of the BP check. Not the other way around.

Reply:
I think we really need to run some kind of wiki for SS13 code structures functions and shit because we are guessing what works and what not.

Reply:
The code understanding page is not enough.

Reply:
It's because there is a chance the BP check will fail, so less processing will potentially be done than if you do the gender check first.

Reply:
I understood this already, but still my idea might be good. I will check some api wiki templates, maybe we'll get one and start filling it up. It will help coding severely anyway.

Reply:
Also reduces copypasta.

Reply:
Yea, because people will understand the code themselves and not copy shitcode from another places.